usaspending_api/reporting/v2/views/submission_history.py

```diff
@@ -0,0 +1,40 @@
+from rest_framework.response import Response
+
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, PaginationMixin
+
+from usaspending_api.common.exceptions import UnprocessableEntityException
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.submissions.models import SubmissionAttributes
+
+
+class SubmissionHistory(PaginationMixin, AgencyBase):
+    """Returns submission history of the specified agency for the specified fiscal year and period"""
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/reporting/agencies/agency_code/fiscal_year/fiscal_period/submission_history.md"
+
+    @staticmethod
+    def validate_fiscal_period(fiscal_period):
+        fiscal_period = int(fiscal_period)
+        if fiscal_period < 2 or fiscal_period > 12:
+            raise UnprocessableEntityException(f"fiscal_period must be in the range 2-12")
```

Comment:
This should be moved to `AgencyBase` to share with usaspending_api/reporting/v2/views/differences.py